sm/plugins/tedge_docker_plugin/tedge_docker_plugin.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+set -e
+
+COMMAND="$1"
+IMAGE_NAME="$2"
+IMAGE_TAG="$3"
+
+case "$COMMAND" in
+    prepare)
+        # nothing to do here
+        ;;
+    list)
+        sudo docker image list --format '{"name":"{{.Repository}}","version":"{{.Tag}}"}'
+        ;;
+    install)
+        # Stop all containers using the provided image name
+        containers=sudo docker ps --format "{{.ID}} {{.Image}}" | grep $IMAGE_NAME | awk '{print $1}'
+        for container in $containers
+        do
+            sudo docker stop $container
+        done
+        # Spawn new containers with the provided image name and version to replace the stopped ones
+        sudo docker run $IMAGE_NAME:$IMAGE_TAG
```

Comment:
When $IMAGE_TAG is empty this command will fail:
![image](https://user-images.githubusercontent.com/75632179/134651104-61bfb7a3-ed6e-436f-93d9-01dfc51de4dd.png)